File: src-tauri/src/commands/project_symlinks.rs
use tauri::State;

use nexus_core::{error::AppResult, services::project_symlinks::ProjectSymlink};

use crate::store::AppState;

#[tauri::command]
pub fn list_project_symlinks(state: State<'_, AppState>) -> AppResult<Vec<ProjectSymlink>> {
    state.project_symlinks.list_project_symlinks()
}

#[tauri::command]
pub fn delete_project_symlink(state: State<'_, AppState>, target_path: String) -> AppResult<()> {
    state.project_symlinks.delete_project_symlink(target_path)
}

File: src-react/src/lib/api/projectSymlinks.ts
import type { ProjectSymlink } from "../../types/index.js";
import { invokeCommand } from "./tauri.js";

export const projectSymlinksApi = {
  list(): Promise<ProjectSymlink[]> {
    return invokeCommand<ProjectSymlink[]>("list_project_symlinks");
  },

  delete(targetPath: string): Promise<void> {
    return invokeCommand<void>("delete_project_symlink", { targetPath });
  },
};
